### PR TITLE
Handle errors in pkg/downwardmetrics

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Ignore error when closing a read-only file.

**Release note**:
```release-note
None
```

### CODE REVIEW

File: pkg/downwardmetrics/vhostmd/disk.go [hunk 1]
```diff
@@ -102,7 +102,10 @@ func readDisk(filePath string) (*Disk, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer f.Close()
+	// It is safe to ignore error that may happen when closing a file, because it is only used for reading.
+	// If the read operation succeeds, but close fails, we have already read the data, so it is safe to ignore.
+	defer func() { _ = f.Close() }()
```
Comment (xpivarc) [inline pkg/downwardmetrics/vhostmd/disk.go:107]: We have `CloseIOAndCheckErr` that can be used.

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
Remove the hold once the comment is addressed.

File: pkg/downwardmetrics/vhostmd/disk.go (hunk 1 above)
Comment (akrejcir) [reply pkg/downwardmetrics/vhostmd/disk.go]: That's useful function, thanks.

**Comment by rmohr** [comment]
@akrejcir please don't mass-create individual trivial tiny PRs which are closely related. Just separate them by commit and bundle the changes at least until a reasonable PR since. As the project is set up, this creates a lot of email, review and CI overhead without benefit.

In the past we still merged such mass PRs, despite being discouraged, but I want to ask you to not continue doing that.

The changes as such are very welcome.

**Comment by akrejcir** [comment]
Closing, and will merge with #8635 

/close

**Pull request closed without merging**